### PR TITLE
Mieubrisse/tighten fully connected test

### PR DESCRIPTION
**Pull request by mieubrisse**
NOTE: Review after https://github.com/kurtosis-tech/ava-e2e-tests/pull/66 !!!

NOTE: This test isn't passing, but given my understanding of how stakers work I think it should be - I want to discuss with you (either here or on Slack) why.

Namely:
```
after we run GetFundsAndStartValidating, do we expect a node to be equivalent to the bootstrappers in that, it has peers == all the other nodes on the network? I'm making the fully-connected test actually test the thing it's supposed to by:
1) spinning up 5 boot nodes (normal)
2) adding two non-boot nodes
3) calling GetFundsAndStartValidating from the highlevel API
4) verifying that the node who became a validator has all 6 other nodes in the network (just like the boot node).
However, this isn't happening - that first node who becomes a validator is still only having the 5 boot nodes in his peer list. This is wrong, correct?
```

### CODE REVIEW

File: commons/ava_testsuite/staking_gecko_network_tests.go [hunk 1]
```diff
@@ -125,159 +120,70 @@ func (test StakingNetworkRpcWorkflowTest) GetTimeout() time.Duration {
...
+
+	normalServiceClient1 := allGeckoClients[normalServiceId1]
 	highLevelExtraStakerClient := ava_networks.NewHighLevelGeckoClient(
-		extraStakerClient,
+		normalServiceClient1,
 		STAKER_USERNAME,
 		STAKER_PASSWORD)
-	err = highLevelExtraStakerClient.GetFundsAndStartValidating(SEED_AMOUNT, STAKE_AMOUNT)
-	if err != nil {
+	if err := highLevelExtraStakerClient.GetFundsAndStartValidating(SEED_AMOUNT, STAKE_AMOUNT); err != nil {
```
Comment (galenmarchetti) [inline commons/ava_testsuite/staking_gecko_network_tests.go:148]: @mieubrisse I believe we'll have to add _both_ non-bootstrapper nodes as  validators in order for this to work

Comment (galenmarchetti) [reply commons/ava_testsuite/staking_gecko_network_tests.go:148]: actually, could be wrong. not sure whether a non-staker node will participate in gossiping to grab more peers. but for sure a validator node participates in the gossip and thus picks up peers

**Comment by galenmarchetti** [comment]
I'll wait on this one until we figure out the tests 👍

File: commons/ava_testsuite/staking_gecko_network_tests.go [hunk 2]
```diff
@@ -123,73 +122,58 @@ func (test StakingNetworkRpcWorkflowTest) GetTimeout() time.Duration {
...
+	if err := verifyNetworkFullyConnected(allServiceIds, stakerIds, allNodeIds, allGeckoClients); err != nil {
+		context.Fatal(stacktrace.Propagate(err, "An error occurred verifying that the network is fully connected after gossip"))
 	}
 }
 
 func (test StakingNetworkFullyConnectedTest) GetNetworkLoader() (testsuite.TestNetworkLoader, error) {
 	return getStakingNetworkLoader(map[int]int{
-		NODE_SERVICE_ID:           NORMAL_NODE_CONFIG_ID,
+		// TODO Once this is split into its own file, make these constants
+		0: NORMAL_NODE_CONFIG_ID,
```
Comment (galenmarchetti) [inline commons/ava_testsuite/staking_gecko_network_tests.go:170]: @mieubrisse these numbers (0, 1) are recorded twice here. once as a variable scoped to the Run function, and once here.

previously we had resolved this by having them as module-wide constants. in my PR on byzantine behavior I see why thats not ideal. however this strategy has this duplication. is there another option available to us to scope these values across the functions here? can we store them as values in the StakingNetworkFullyConnectedTest struct, for example?

File: commons/ava_testsuite/staking_gecko_network_tests.go (hunk 1 above)
Comment (mieubrisse) [reply commons/ava_testsuite/staking_gecko_network_tests.go:147]: Yep, we don't need to

File: commons/ava_testsuite/staking_gecko_network_tests.go (hunk 2 above)
Comment (galenmarchetti) [reply commons/ava_testsuite/staking_gecko_network_tests.go:170]: based on slack conversation we're going to handle this in another PR

**Pull request merged**